### PR TITLE
feat(python): add support for uv dev and optional dependencies

### PR DESCRIPTION
**Pull request by nikpivkin**
## Description

This PR adds support for optional and dev dependencies. Optional dependencies are handled the same way for poetry. The optional dependencies of the root package are added as `direct` and the rest as `indirect`.

### Without `--include-dev-deps`
```bash
❯ ./trivy fs /Users/nikita/projects/trivy-test/uv-test -q --list-all-pkgs -f json | jq '.Results[] | .Packages[] | select(.Name == "pytest")'
```

### With`--include-dev-deps`
```bash
❯ ./trivy fs /Users/nikita/projects/trivy-test/uv-test -q --list-all-pkgs -f json --include-dev-deps | jq '.Results[] | .Packages[] | select(.Name == "pytest")'
{
  "ID": "pytest@8.3.4",
  "Name": "pytest",
  "Identifier": {
    "PURL": "pkg:pypi/pytest@8.3.4",
    "UID": "5ccb63bab9c8e426"
  },
  "Version": "8.3.4",
  "Dev": true,
  "Relationship": "direct",
  "DependsOn": [
    "colorama@0.4.6",
    "iniconfig@2.0.0",
    "packaging@24.2",
    "pluggy@1.5.0"
  ],
  "Layer": {}
}
```

`pyptoject.toml`:
```toml
[project]
name = "uv-test"
version = "0.1.0"
description = "Add your description here"
readme = "README.md"
requires-python = ">=3.12"

[dependency-groups]
test = [
    "pytest>=8.3.4",
]

```

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/dependency/parser/python/uv/parse_testcase.go [hunk 1]
```diff
@@ -9,23 +9,44 @@ var (
...
+	// uv add httpx==0.28.1 --extra socks
+	// uv add orjson==3.10.12 --optional json
 	// apk add jq
 	// uv pip list --format json |jq -c 'sort_by(.name) | .[] | {"ID": (.name + "@" + .version), "Name": .name, "Version": .version}' | sed 's/$/,/' | sed 's/\"\([^"]*\)\":/\1:/g'
 
 	// add a root project
 	// fill in the relationships between the packages
 	uvNormal = []ftypes.Package{
 		{ID: "normal@0.1.0", Name: "normal", Version: "0.1.0", Relationship: ftypes.RelationshipRoot},
+		{ID: "httpx@0.28.1", Name: "httpx", Version: "0.28.1", Relationship: ftypes.RelationshipDirect},
```
Comment (knqyf263) [inline pkg/dependency/parser/python/uv/parse_testcase.go:21]: This package is not marked as a development dependency. Is it correct?

Comment (nikpivkin) [reply pkg/dependency/parser/python/uv/parse_testcase.go:21]: `httpx` is not a development dependency: `uv add httpx==0.28.1 --extra socks`

Comment (knqyf263) [reply pkg/dependency/parser/python/uv/parse_testcase.go:21]: Why did we newly introduce this dependency? I thought the test case was updated for optional or development dependencies.

Comment (knqyf263) [reply pkg/dependency/parser/python/uv/parse_testcase.go:21]: Did we need it to test extra packages?

Comment (nikpivkin) [reply pkg/dependency/parser/python/uv/parse_testcase.go:21]: Ah, I should have added about test cases in the description. Yes, I added some more test cases:
- An optional dependency in the root package
- Direct dependency with an extra dependency that is optional

Comment (knqyf263) [reply pkg/dependency/parser/python/uv/parse_testcase.go:21]: OK, thanks.

**Comment by knqyf263** [comment]
@nikpivkin Can you rebase the main branch? I think the linter will fail on `map[string]struct{}`.

File: docs/docs/coverage/language/python.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ The following table provides an outline of the features Trivy offers.
 | pip             | requirements.txt |            -            |     Include      |                  -                   |    ✓     |                    ✓                     |
 | Pipenv          | Pipfile.lock     |            ✓            |     Include      |                  -                   |    ✓     |                Not needed                |
 | Poetry          | poetry.lock      |            ✓            |     Exclude      |                  ✓                   |    -     |                Not needed                |
-| uv              | uv.lock          |            ✓            |     Exclude      |                  ✓                   |    -     |                Not needed                |
+| uv              | uv.lock          |            ✓            |     Include      |                  ✓                   |    -     |                Not needed                |
```
[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply docs/docs/coverage/language/python.md:30]: Can use link for `Include` as in [nodejs page](https://trivy.dev/latest/docs/coverage/language/nodejs/)?

**Comment by nikpivkin** [comment]
I'm working on it right now

File: docs/docs/coverage/language/python.md [hunk 3]
```diff
@@ -104,6 +104,8 @@ urllib3==1.26.15
 `requirements.txt` files don't contain information about dependencies used for development.
 Trivy could detect vulnerabilities on the development packages, which not affect your production environment.
 
+By default, Trivy doesn't report development dependencies. Use the `--include-dev-deps` flag to include them.
```
Comment (DmitriyLewen) [inline docs/docs/coverage/language/python.md:107]: Trivy doesn't support `--include-dev-deps` flag for `pip`

File: docs/docs/coverage/language/python.md [hunk 4]
```diff
@@ -122,6 +124,8 @@ Trivy could detect vulnerabilities on the development packages, which not affect
 
 License detection is not supported for `Pipenv`.
 
+By default, Trivy doesn't report development dependencies. Use the `--include-dev-deps` flag to include them.
```
Comment (DmitriyLewen) [inline docs/docs/coverage/language/python.md:127]: same

Comment (nikpivkin) [reply docs/docs/coverage/language/python.md]: Fixed https://github.com/aquasecurity/trivy/pull/8134/commits/55fb02f3be0c5a663afc943eb822a0b0a1872ac8

File: docs/docs/coverage/language/python.md (hunk 3 above)
Comment (nikpivkin) [reply docs/docs/coverage/language/python.md]: Fixed https://github.com/aquasecurity/trivy/pull/8134/commits/55fb02f3be0c5a663afc943eb822a0b0a1872ac8

Comment (nikpivkin) [reply docs/docs/coverage/language/python.md]: > Can use link for Include as in [nodejs page](https://trivy.dev/latest/docs/coverage/language/nodejs/)?
By the way, why are `include/exclude` used to link the table and sections and not the package manager names from the first column?

Comment (DmitriyLewen) [reply docs/docs/coverage/language/python.md]: By default, we don't show Dev dependencies. 
This link is intended to draw users attention to this and direct them to the section about the file, where they can see that `--include-dev-deps` flag needs to be used.

File: docs/docs/coverage/language/python.md [hunk 5]
```diff
@@ -24,10 +24,10 @@ The following table provides an outline of the features Trivy offers.
...
-| Poetry          | poetry.lock      |            ✓            |     Exclude      |                  ✓                   |    -     |                Not needed                |
-| uv              | uv.lock          |            ✓            |     Exclude      |                  ✓                   |    -     |                Not needed                |
+| pip             | requirements.txt |            -            |     [Include](#pip)      |                  -                   |    ✓     |                    ✓                     |
+| Pipenv          | Pipfile.lock     |            ✓            |     [Include](#pipenv)      |                  -                   |    ✓     |                Not needed                |
+| Poetry          | poetry.lock      |            ✓            |     [Exclude](#poetry)      |                  ✓                   |    -     |                Not needed                |
+| uv              | uv.lock          |            ✓            |     [Include](#uv)      |                  ✓                   |    -     |                Not needed                |
```
Comment (DmitriyLewen) [inline docs/docs/coverage/language/python.md:30]: you only need to add links if the `--include-dev-deps` flag is used for the file

Comment (nikpivkin) [reply docs/docs/coverage/language/python.md:30]: In nodejs all package managers have links. I think the documentation should be consistent.

Comment (nikpivkin) [reply docs/docs/coverage/language/python.md:30]: Then we should update the nodejs documentation

Comment (nikpivkin) [reply docs/docs/coverage/language/python.md]: Fixed

Comment (DmitriyLewen) [reply docs/docs/coverage/language/python.md]: > In nodejs all package managers have links. I think the documentation should be consistent.

I missed that a link was added for `Bun`. We can remove the link for that.
But for other files - `yarn`, `npm` and `pnpm` support `--include-dev-deps`. That's why they have links.

File: docs/docs/coverage/language/python.md (hunk 2 above)
Comment (knqyf263) [inline docs/docs/coverage/language/python.md]: I think we should write the default behavior, like [Node.js](https://github.com/aquasecurity/trivy/blob/03647db96f36b2aba766f65fb0c213dbf937ae62/docs/docs/coverage/language/nodejs.md). Or am I missing something?

```suggestion
| uv              | uv.lock          |            ✓            |     Exclude      |                  ✓                   |    -     |                Not needed                |
```

File: docs/docs/coverage/language/python.md (hunk 5 above)
Comment (nikpivkin) [reply docs/docs/coverage/language/python.md]: OK, I'll open a PR for that.

File: docs/docs/coverage/language/python.md (hunk 2 above)
Comment (nikpivkin) [reply docs/docs/coverage/language/python.md]: Fixed https://github.com/aquasecurity/trivy/pull/8134/commits/5d65e2f6c6dd6776667728a230866f1dd0dd14b5

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**